### ISSUE TITLE
Undefined Method removeFolder in ImapEmailFolderRepository

### ISSUE THREAD

**Issue by Deltachaos** [orocrm/platform#419]
I have seen that there is a usage of the removeFolder in ImapEmailFolderManager, but the repository ImapEmailFolderRepository not contains this method.

```
exception 'BadMethodCallException' with message 'Undefined method 'removeFolder'. The method name must start with either findBy or findOneBy!' in /var/www/crm/vendor/doctrine/orm/lib/Doctrine/ORM/EntityRepository.php:226
Stack trace:
#0 /var/www/crm/vendor/oro/platform/src/Oro/Bundle/ImapBundle/Manager/ImapEmailFolderManager.php(116): Doctrine\ORM\EntityRepository->__call('removeFolder', Array)
#1 /var/www/crm/vendor/oro/platform/src/Oro/Bundle/ImapBundle/Manager/ImapEmailFolderManager.php(116): Oro\Bundle\ImapBundle\Entity\Repository\ImapEmailFolderRepository->removeFolder(Object(Oro\Bundle\ImapBundle\Entity\ImapEmailFolder))
#2 /var/www/crm/vendor/oro/platform/src/Oro/Bundle/ImapBundle/Manager/ImapEmailFolderManager.php(70): Oro\Bundle\ImapBundle\Manager\ImapEmailFolderManager->removeOutdated(Object(Doctrine\Common\Collections\ArrayCollection))
#3 /var/www/crm/vendor/oro/platform/src/Oro/Bundle/ImapBundle/Controller/ConnectionController.php(72): Oro\Bundle\ImapBundle\Manager\ImapEmailFolderManager->getFolders()
#4 [internal function]: Oro\Bundle\ImapBundle\Controller\ConnectionController->checkAction(Object(Symfony\Component\HttpFoundation\Request))
#5 /var/www/crm/app/bootstrap.php.cache(3109): call_user_func_array(Array, Array)
#6 /var/www/crm/app/bootstrap.php.cache(3071): Symfony\Component\HttpKernel\HttpKernel->handleRaw(Object(Symfony\Component\HttpFoundation\Request), 1)
#7 /var/www/crm/app/bootstrap.php.cache(3222): Symfony\Component\HttpKernel\HttpKernel->handle(Object(Symfony\Component\HttpFoundation\Request), 1, true)
#8 /var/www/crm/app/bootstrap.php.cache(2444): Symfony\Component\HttpKernel\DependencyInjection\ContainerAwareHttpKernel->handle(Object(Symfony\Component\HttpFoundation\Request), 1, true)
#9 /var/www/crm/web/app.php(22): Symfony\Component\HttpKernel\Kernel->handle(Object(Symfony\Component\HttpFoundation\Request))
#10 {main}
```

**Comment by mccar** [comment]
Hi Deltachaos,

Thanks for pointing this issue, we will fix this it in one of the future releases.

**Comment by Deltachaos** [comment]
@mccar seems like this has been solved by 6051888f2aae89633f99f1a7ecbdc39130ec82c8
